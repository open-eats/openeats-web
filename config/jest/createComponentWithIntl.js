import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

// See: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#jest

const createComponentWithIntl = (children, props = { locale: 'en' }) => {
  return renderer.create(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  );
};

export default createComponentWithIntl;