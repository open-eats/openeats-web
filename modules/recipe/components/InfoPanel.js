import React from 'react'
import PropTypes from 'prop-types'
import {
    injectIntl,
    defineMessages,
} from 'react-intl'

import { Input } from '../../common/components/FormComponents'

const RecipeHeader = ({ cookTime, prepTime, servings, customServings, info, updateServings, clearServings, intl }) => {
  const messages = defineMessages({
    servings: {
      id: 'recipe.servings',
      description: 'Servings',
      defaultMessage: 'Servings',
    },
    prep_time: {
      id: 'recipe.prep_time',
      description: 'Preparation time',
      defaultMessage: 'Prep time',
    },
    cooking_time: {
      id: 'recipe.cooking_time',
      description: 'Cooking time',
      defaultMessage: 'Cooking time',
    },
    minutes: {
      id: 'recipe.minutes',
      description: 'minutes',
      defaultMessage: 'minutes'
    },
  });

  let clearInput = '';
  if (servings !== customServings) {
    clearInput = (
      <span className="input-group-btn">
        <button className="btn btn-default" type="button" onClick={ clearServings.bind(this, servings) }>
          <span className="glyphicon glyphicon-remove" aria-hidden="true"/>
        </button>
      </span>
    )
  }

  return (
    <div className="panel panel-default">
      <table className="table table-bordered">
        <thead>
          <tr className="active">
            <th>{ intl.formatMessage(messages.servings) }</th>
            <th>{ intl.formatMessage(messages.prep_time) }</th>
            <th>{ intl.formatMessage(messages.cooking_time) }</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <div className="input-group print-hidden">
                <Input
                  name="servings"
                  type="number"
                  size="servings-textbox"
                  change={ updateServings }
                  value={ customServings!==0 ? customServings : '' } />
                { clearInput }
              </div>
              <p className="print-only">{ customServings ? customServings : servings }</p>
            </td>
            <td>{ prepTime } { intl.formatMessage(messages.minutes) }</td>
            <td>{ cookTime } { intl.formatMessage(messages.minutes) }</td>
          </tr>
        </tbody>
      </table>
      <div className="panel-body">
        <p>{ info }</p>
      </div>
    </div>
  );
};

RecipeHeader.PropTypes = {
  cookTime: PropTypes.number.isRequired,
  prepTime: PropTypes.number.isRequired,
  servings: PropTypes.number.isRequired,
  customServings: PropTypes.string,
  info: PropTypes.string.isRequired,
  updateServings: PropTypes.func.isRequired,
  clearServings: PropTypes.func.isRequired,
  intl: PropTypes.object.isRequired,
};

export default injectIntl(RecipeHeader);
