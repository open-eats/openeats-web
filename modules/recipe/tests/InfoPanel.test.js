import React from 'react';
import InfoPanel from '../components/InfoPanel';
import createComponentWithIntl from '../../../config/jest/createComponentWithIntl';

test('Short Info Panel', () => {
  const mockUpdateServings = jest.fn();
  const mockClearServings = jest.fn();
  const component = createComponentWithIntl(
    <InfoPanel
        cookTime={ 20 }
        prepTime={ 40 }
        servings={ 8 }
        info={ 'This is a test!' }
        updateServings={ mockUpdateServings }
        clearServings={ mockClearServings }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Long Info Panel', () => {
  const mockUpdateServings = jest.fn();
  const mockClearServings = jest.fn();
  const longString = 'Lorem ipsum dolor sit amet, odio sem vel ' +
      'nulla aliquam, vestibulum pellentesque at risus eros, at ' +
      'leo mi urna quam, magna nunc vestibulum nibh condimentum ' +
      'arcu. Eu nunc blandit arcu magna, morbi eleifend dolor, nam ' +
      'ratione justo ullamcorper platea malesuada donec, fusce' +
      ' luctus varius. Vel ipsum nunc in justo, pulvinar magna ' +
      'interdum ut nunc vestibulum. Lorem dui vitae. Lectus elit vel' +
      ' lorem purus, in tortor, mi diam mus venenatis sit bibendum eu.' +
      ' Odio nec lectus quisquam ac eleifend sem, arcu congue neque neque ' +
      'nunc, nam maecenas, nec mauris pede erat dignissim, in gravida tempor.' +
      ' In amet mauris odio vitae montes. Nec vitae nostrud vestibulum, ' +
      'at in tincidunt justo varius wisi ut, nullam wisi erat ' +
      'lorem tempor ac. Lorem ipsum dolor sit amet, odio sem vel ' +
      'nulla aliquam, vestibulum pellentesque at risus eros, at ' +
      'leo mi urna quam, magna nunc vestibulum nibh condimentum ' +
      'arcu. Eu nunc blandit arcu magna, morbi eleifend dolor, nam ' +
      'ratione justo ullamcorper platea malesuada donec, fusce' +
      ' luctus varius. Vel ipsum nunc in justo, pulvinar magna ' +
      'interdum ut nunc vestibulum. Lorem dui vitae. Lectus elit vel' +
      ' lorem purus, in tortor, mi diam mus venenatis sit bibendum eu.' +
      ' Odio nec lectus quisquam ac eleifend sem, arcu congue neque neque ' +
      'nunc, nam maecenas, nec mauris pede erat dignissim, in gravida tempor.' +
      ' In amet mauris odio vitae montes. Nec vitae nostrud vestibulum, ' +
      'at in tincidunt justo varius wisi ut, nullam wisi erat ' +
      'lorem tempor ac.';
  const component = createComponentWithIntl(
    <InfoPanel
        cookTime={ 200000 }
        prepTime={ 200000 }
        servings={ 200000 }
        info={ longString }
        updateServings={ mockUpdateServings }
        clearServings={ mockClearServings }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
