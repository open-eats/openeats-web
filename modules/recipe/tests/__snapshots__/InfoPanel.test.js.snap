// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Long Info Panel 1`] = `
<div
  className="panel panel-default"
>
  <table
    className="table table-bordered"
  >
    <thead>
      <tr
        className="active"
      >
        <th>
          Servings
        </th>
        <th>
          Prep time
        </th>
        <th>
          Cooking time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            className="input-group print-hidden"
          >
            <div
              className="servings-textbox"
            >
              <div
                className="form-group null"
              >
                <input
                  className="form-control"
                  name="servings"
                  onChange={[Function]}
                  placeholder={undefined}
                  type="number"
                  value=""
                />
              </div>
            </div>
            <span
              className="input-group-btn"
            >
              <button
                className="btn btn-default"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                  className="glyphicon glyphicon-remove"
                />
              </button>
            </span>
          </div>
          <p
            className="print-only"
          >
            200000
          </p>
        </td>
        <td>
          200000
           
          minutes
        </td>
        <td>
          200000
           
          minutes
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="panel-body"
  >
    <p>
      Lorem ipsum dolor sit amet, odio sem vel nulla aliquam, vestibulum pellentesque at risus eros, at leo mi urna quam, magna nunc vestibulum nibh condimentum arcu. Eu nunc blandit arcu magna, morbi eleifend dolor, nam ratione justo ullamcorper platea malesuada donec, fusce luctus varius. Vel ipsum nunc in justo, pulvinar magna interdum ut nunc vestibulum. Lorem dui vitae. Lectus elit vel lorem purus, in tortor, mi diam mus venenatis sit bibendum eu. Odio nec lectus quisquam ac eleifend sem, arcu congue neque neque nunc, nam maecenas, nec mauris pede erat dignissim, in gravida tempor. In amet mauris odio vitae montes. Nec vitae nostrud vestibulum, at in tincidunt justo varius wisi ut, nullam wisi erat lorem tempor ac. Lorem ipsum dolor sit amet, odio sem vel nulla aliquam, vestibulum pellentesque at risus eros, at leo mi urna quam, magna nunc vestibulum nibh condimentum arcu. Eu nunc blandit arcu magna, morbi eleifend dolor, nam ratione justo ullamcorper platea malesuada donec, fusce luctus varius. Vel ipsum nunc in justo, pulvinar magna interdum ut nunc vestibulum. Lorem dui vitae. Lectus elit vel lorem purus, in tortor, mi diam mus venenatis sit bibendum eu. Odio nec lectus quisquam ac eleifend sem, arcu congue neque neque nunc, nam maecenas, nec mauris pede erat dignissim, in gravida tempor. In amet mauris odio vitae montes. Nec vitae nostrud vestibulum, at in tincidunt justo varius wisi ut, nullam wisi erat lorem tempor ac.
    </p>
  </div>
</div>
`;

exports[`Short Info Panel 1`] = `
<div
  className="panel panel-default"
>
  <table
    className="table table-bordered"
  >
    <thead>
      <tr
        className="active"
      >
        <th>
          Servings
        </th>
        <th>
          Prep time
        </th>
        <th>
          Cooking time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            className="input-group print-hidden"
          >
            <div
              className="servings-textbox"
            >
              <div
                className="form-group null"
              >
                <input
                  className="form-control"
                  name="servings"
                  onChange={[Function]}
                  placeholder={undefined}
                  type="number"
                  value=""
                />
              </div>
            </div>
            <span
              className="input-group-btn"
            >
              <button
                className="btn btn-default"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                  className="glyphicon glyphicon-remove"
                />
              </button>
            </span>
          </div>
          <p
            className="print-only"
          >
            8
          </p>
        </td>
        <td>
          40
           
          minutes
        </td>
        <td>
          20
           
          minutes
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="panel-body"
  >
    <p>
      This is a test!
    </p>
  </div>
</div>
`;
